content/blog/reducing-openapi.md

```diff
@@ -0,0 +1,252 @@
+---
+date: 2024-09-05
+title: Creating Reduced OpenAPI Documents for Integrating Systems on Serverless Workflows
+---
+
+# Creating Reduced OpenAPI Documents for Integrating Systems on Serverless Workflows
+The blog post will guide developers on how to reduce openAPI documents to a manageable size. The need for this procedure has risen in account of restrictions that Quarkus imposes with accepting large YAML files as input [(see appendix)](#appendix). This restriction directs us to be mindful and plan ahead which resources services we would need in our workflow. 
+
+
+In this guide, we will explain what is an OpenAPI Document, how to read and use the openAPI Specification, and eventually we will cover the steps to reduce an openAPI document in a valid manner. 
+
+## What is an OpenAPI Document?
+>A self-contained or composite resource which defines or describes an API or elements of an API. 
+
+OpenAPI documents are a standardized way to view a system’s exposed paths, resources and webhooks, and act as an alternative to using restAPI calls for interacting with other systems.
+An OpenAPI document uses and conforms to the OpenAPI Specification, and in itself is a JSON object (which may be represented either in JSON or YAML format). 
+
+For the OpenAPI Spec Documentation, see: https://swagger.io/specification/
+
+## How to make sense of an OpenAPI Document:
+Let’s take a look at the [aap-openAPI document](https://github.com/parodos-dev/serverless-workflow-examples/blob/main/aap-db-deploy-quarkus/src/main/resources/specs/aap-openapi.yaml) for some reference: 
```

Comment:
1. To avoid possible broken link, let's include the referred files in this repository.
2. Let's use https://github.com/parodos-dev/serverless-workflow-examples/blob/main/aap-db-deploy-quarkus/src/main/resources/specs/github-openapi.yaml  instead of the aap-openapi.yaml
It contains a recursive reference of components and is a reduced version of https://github.com/github/rest-api-description/blob/main/descriptions/api.github.com/api.github.com.yaml
so we can mention both (the start and the end point of this blog)